Backend/src/main/java/com/harmonia/backend/controller/DirectMessageController.java

```diff
@@ -51,6 +87,12 @@ public ResponseEntity<DirectMessage> sendDMessage(@RequestBody DMessageRequest d
         }
     }
 
+    /**
+     * Handles DELETE requests to delete a direct message.
+     *
+     * @param directMessage a DirectMessage object representing the direct message to delete
+     * @return a ResponseEntity containing a message indicating that the direct message was deleted, or a BAD_REQUEST status if an error occurred
+     */
     @DeleteMapping
     @CrossOrigin
     public ResponseEntity<String> deleteDMessage(@RequestBody DirectMessage directMessage) {
```

Comment:
@DeleteMapping("dmessageId/{dmessageId}")
Deletes are always done by the primary key.